### DISCUSSION TITLE
TFT70 bootloader

### DISCUSSION THREAD

**Discussion by paulpenney** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1596] [General]
Can you guys please make the TFT70 bootloader available? 

Thanks,

Paul

**Comment by fvanroie** [comment]
I have a bricked TFT70 that I'm trying to reflash...
Where can I find the bootloader needed for the device please?

**Comment by radek8** [comment]
Were you able to download the botloader for TFT70?
If so, can you share it for other community users?
Thank you

**Comment by jafarniknam** [comment]
hello 
I also need to make changes to the 7-inch display because I use for custom robot that isn't a 3D Printer . Is there any way to upload BTT FW on mks tft 70 displays? 

can you help me please .
@darkspr1te

**Reply by darkspr1te** [reply]
yes there is but you will have to adjust the pinouts file and screen type to match the 7inch display and what ever cpu the 7" runs off. I dont have one so i cant confirm any details.

**Comment by aktronicsdotnet** [comment]
Hello everyone,
I put this file over here just in case anyone else has bricked their BIGTREETECH TFT70,

I extracted this bootloader from my own board using ST-Link programmer
You will need a programmer too

First erase the MCU completely then program the attached file at address 0x08000000
After it's done you should see something like "Illegal app" on display.
Grab the latest firmware from [https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/tree/master/Copy%20to%20SD%20Card%20root%20directory%20to%20update](url) and copy it to a formatted SD card
Connect the SD card to your board while it's off, Power the board up. You'll see the update message.
After firmware update is done you might need to update the graphics too.
Grab them from the same folder I linked, Copy TFT70 folder and Config.ini to your SD card and update your board.

After you are done with your board, your display should be usable again.

[BIGTREETECH TFT70 V3.0 STM32F407VG Bootloader.zip](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/8888393/BIGTREETECH.TFT70.V3.0.STM32F407VG.Bootloader.zip)

**Reply by ArtemShchuchev** [reply]
Thank you so much for the bootloader! I accidentally deleted it from my display. Thanks to your post, everything is restored! BIGTREETECH TFT7.0 V3.0: CPU - GD32F407VET6

**Reply by radek8** [reply]
Did it work for you on the GD32 processor variant?

**Reply by ArtemShchuchev** [reply]
> 
Yes
Moreover, unlike the native one, this one works well. The native bootloader did not work correctly because of this, I began to deal with it and unknowingly deleted it. I was very upset, but now everything is fine! 
![20220619_194712](https://user-images.githubusercontent.com/107797299/174481764-108f0ffb-1920-4988-ae9d-083cf1bdbe15.jpg)

**Reply by Rodolfo131** [reply]
Hi Artem
I have he same problem and board, could you show me where is the st link connected. Thank you

**Reply by ArtemShchuchev** [reply]
Hi Rodolfo131
![1-TFT50-pinout](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/assets/107797299/a2798b8c-e49d-4e75-b750-e06a4b4bea11)

**Comment by radek8** [comment]
Hi @skopusdotorg 
That's great. It will definitely help many people.
I can share your post here:
https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/tree/master/Bootloaders
to make it more accessible to more people?
Thank you

**Reply by aktronicsdotnet** [reply]
@radek8
That would be nice but I think first someone with a bricked TFT needs to confirm that this bootloader works on their board too.
Sometimes manufacturers lock the bootloader to a specific MCU using internal serial number of the MCU, If someone confirm that this bootloader is not locked only to my board then you can share it there for everyone to use...

**Comment by radek8** [comment]
Bootloaders for 4 TFT models from BTT have already been published here and have never been locked.
I guess it won't be locked for the TFT70 either.
Sometimes there is someone who destroys the processor on their display, is able to replace the processor and then ends up with no bootlooader.
This will help him a lot.

**Comment by radek8** [comment]
@fvanroie , @paulpenney ,
Did you manage to fix your bootloader on Btt TFT70?

**Reply by fvanroie** [reply]
Unfortunately not, BTT has been dead silent on this issue...

**Reply by darkspr1te** [reply]
Maybe i can help, i re-wrote bootloaders for most models and could possibly help you trace your issue with debug code.

**Reply by radek8** [reply]
> Unfortunately not, BTT has been dead silent on this issue...

@fvanroie 
You can try bootloader from
skopusdotorg?

**Reply by aktronicsdotnet** [reply]
@fvanroie I can confirm the file I shared on any MCU so you can easily reprogram your board using the method I shared. Manufacturer never give you something like a bootloader. When I asked for me ender 3 v2 mainboard bootloader but they refused to share it too.
@radek8 You can share the file in the main repository...

**Comment by radek8** [comment]
https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/pull/2543

**Comment by ArtemShchuchev** [comment]
Hello Henryking39!

Unfortunately, I can’t help with anything, because. I'm not good enough at this. I initially had a bad bootloader and during the experiments I killed it by turning the display into a brick. So I had nothing to lose. I downloaded the bootloader from this person's post "skopusdotorg on 13 Jun", while I had a Chinese processor, not STM. And I was lucky, everything worked.

You can try asking "radek8" or "skopusdotorg" from this thread "https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/discussions/1596#discussioncomment-3163364".
You can also write ***@***.***" - this is BIGTREETECH technical support.

I'm sorry. Good luck!

Best regards,
Artem

От: Henryking39
Отправлено: 17 июля 2022 г. в 3:43
Кому: bigtreetech/BIGTREETECH-TouchScreenFirmware
Копия: ArtemShchuchev; Comment
Тема: Re: [bigtreetech/BIGTREETECH-TouchScreenFirmware] TFT70 bootloader(Discussion #1596)

Hello. On the downloads page, I saw that the TFT 70 V2 is not included. Is there any way I can download the installation files? Since mine is V2 (no button, just touch) and not V3.0
If anyone can help me I would be very grateful.
Thanks!
Below is the list of supported displays.
Supported Screens
Only the TFTs listed below are currently supported. Trying to install the firmware on a TFT which is not supported can harm the hardware.
BTT TFT
BTT_TFT24_V1.1
BTT_TFT28_V1.0 and V3.0
BTT_TFT35_V1.0, V1.1, V1.2, V2.0, V3.0, E3_V3.0 and B1_V3.0
BTT_TFT43_V3.0
BTT_TFT50_V3.0
BTT_TFT70_V3.0
BTT GD TFT
BTT_GD_TFT24_V1.1
BTT_GD_TFT35_V2.0, V3.0, E3_V3.0 and B1_V3.0
BTT_GD_TFT43_V3.0
BTT_GD_TFT50_V3.0
BTT_GD_TFT70_V3.0
MKS TFT
MKS_TFT28_V3.0 and V4.0
MKS_TFT32_V1.3 and V1.4
MKS_TFT35_V1_0
WARNING: BTT does not officially provide MKS TFT hardware support. MKS TFT is maintained by open source contributors and BTT does not bear any risk of MKS TFT hardware using this firmware.

**Comment by Henryking39** [comment]
Hello Artem. I will try yes. But first, I'm going to ask Bigtreetech support for help. I bought this model because I don't want physical buttons. When I do, I'll give you a feedback. Thank you very much.

**Comment by Rodolfo131** [comment]
Thank you very much.

I was trying to modify to run other app and did something wrong.
I read someone is running  and modifying the software and this is a very
good platform mo work on it.
Let's see if I have  good luck to do.
